YAML/YAML.sublime-syntax

```diff
@@ -1,97 +1,529 @@
+            - match: '{{_flow_scalar_end_plain_in}}'
+              pop: true
+    - include: flow-node
+    - match: :(?=\s|$|{{c_flow_indicator}}) # Empty mapping keys allowed
+      scope: punctuation.separator.mapping.key-value.yaml
+      push: flow-pair-value
+
+  flow-pair-value:
+    - meta_scope: meta.flow-pair-value.yaml
+    - include: flow-node
+    - match: ','
+      scope: punctuation.separator.mapping.pair.yaml
+      pop: true
+    - match: (?=[},\]])
+      pop: true
+
+  block-scalar:
+    # http://www.yaml.org/spec/1.2/spec.html#style/block/scalar
+    # c-l+literal(n) | c-l+folded(n)
+    - match: (?:(\|)|(>))([1-9])?([-+])?  # c-b-block-header(m,t)
       captures:
-        1: entity.name.tag.yaml
-        2: keyword.operator.merge-key.yaml
-        3: punctuation.definition.keyword.yaml
-    - include: comment
-    - match: (#).*?(?=%>)
-    - match: "-"
-      scope: keyword.operator.symbol
+        1: punctuation.definition.block.scalar.literal.yaml
+        2: punctuation.definition.block.scalar.folded.yaml
+        3: constant.numeric.indentation-indicator.yaml
+        4: support.other.chomping-indicator.yaml
+      push:
+        - meta_include_prototype: false
+        - match: ^([ ]+)(?! )  # match first non-empty line to determine indentation level
+          # note that we do not check if indentation is enough
+          set:
+            - meta_scope: string.unquoted.block.yaml
+              meta_include_prototype: false
+            - match: ^(?!\1|\s*$)
+              pop: true
+        - match: ^(?=\S)  # the block is empty
+          pop: true
+        - include: comment  # include comments but not properties
+        - match: .+
+          scope: invalid.illegal.expected-comment-or-newline.yaml
+
+  block-sequence:
+    # http://www.yaml.org/spec/1.2/spec.html#style/block/sequence
+    # l+block-sequence(n)
+    - match: (-)\s  # ( |\t|$) # https://github.com/SublimeTextIssues/Core/issues/963
+      captures:
+        1: punctuation.definition.block.sequence.item.yaml
+
+  block-mapping:
+    # http://www.yaml.org/spec/1.2/spec.html#style/block/mapping
+    # l+block-mapping(n)
+    - include: block-pair
+
+  block-pair:
+    - match: \?
+      scope: punctuation.definition.key-value.begin.yaml
+      push:
+        - meta_scope: meta.block-mapping.yaml
+        - match: (?=\?) # Empty mapping keys & values are allowed
+          pop: true
+        - match: ^ *(:)
+          captures:
+            1: punctuation.separator.mapping.key-value.yaml
+          pop: true
+        - match: ':'
+          scope: invalid.illegal.expected-newline.yaml
+          pop: true
+        - include: block-node
+    # Attempt to match plain-out scalars and highlight as "entitiy.name.tag",
+    # if followed by a colon
+    - match: |
+        (?x)
+        (?=
+          {{ns_plain_first_plain_out}}
+          ((?!{{_flow_scalar_end_plain_out}}).)*
```

Comment:
Here you have a lookahead, containing a negative lookahead, containing a positive lookahead.

Proposed change: `([^ :#]|\:[^ ]| [^#])*`